edge/cmd/edgecore/app/server.go

```diff
@@ -186,8 +186,9 @@ func environmentCheck(skipCheck bool) error {
 		switch processName {
 		case "kubelet": // if kubelet is running, return error
 			return errors.New("kubelet should not running on edge node when running edgecore")
-		case "kube-proxy": // if kube-proxy is running, return error
-			return errors.New("kube-proxy should not running on edge node when running edgecore")
+		case "kube-proxy":
+			// kube-proxy pod does not conflict with edgecore, proceed to restart edgecore
+			klog.Infof("kube-proxy is running when restarting edgecore")
```

Comment:
Now `kube-proxy` can indeed run on edge nodes! And this log will also be printed when edgecore is started for the first time, could users understand what it means? :)

Reply:
Ah, that is true. that does not really explain what to do for user...
So we should do keep it silence? not checking nor printing anything? i guess this sounds fine too.
any thoughts?

Reply:
I think we can skip the kube-proxy check here. Because it can run on edgenode now, plz take a look. cc @WillardHu @Shelley-BaoYue

Reply:
agree on this, fixed in https://github.com/kubeedge/kubeedge/pull/5897/commits/2e8b274056e5724dc47d8e4dee7673af6b04022c